astropy/io/tests/test_registry.py

```diff
@@ -506,12 +505,7 @@ class MTable(Table):
         mt['a'].format = '.4f'
         mt['a'].description = 'hello'
 
-        if HAS_YAML:
-            ctx = nullcontext()
-        else:
-            ctx = pytest.warns(
-                AstropyUserWarning,
-                match="These will be dropped unless you install PyYAML")
+        ctx = nullcontext()
```

Comment:
There is no need for nullcontext anymore. Just remove `with ctx`.

Reply:
Done!   I'll wait for the test suite to pass / fail before pushing